packages/vite/src/node/server/sourcemap.ts

```diff
@@ -5,16 +5,26 @@ export async function injectSourcesContent(
   map: { sources: string[]; sourcesContent?: string[]; sourceRoot?: string },
   file: string
 ) {
-  const sourceRoot = await fs.realpath(
-    path.resolve(path.dirname(file), map.sourceRoot || '')
-  )
+  try {
+    var sourceRoot = await fs.realpath(
+      path.resolve(path.dirname(file), map.sourceRoot || '')
+    )
+  } catch (e) {
+    if (e.code == 'ENOENT') return
+    throw e
+  }
   map.sourcesContent = []
   await Promise.all(
     map.sources.map(async (sourcePath, i) => {
-      map.sourcesContent![i] = await fs.readFile(
-        path.resolve(sourceRoot, decodeURI(sourcePath)),
-        'utf-8'
-      )
+      try {
+        map.sourcesContent![i] = await fs.readFile(
+          path.resolve(sourceRoot, decodeURI(sourcePath)),
+          'utf-8'
+        )
+      } catch (e) {
+        if (e.code == 'ENOENT') return
```

Comment:
Maybe here it's better to log some warning, packages authors should fix this problem by not including sourcemaps in packages.

Reply:
No, sourcemaps should not be discouraged. Sources should be published by npm packages, for easier debugging. The `ENOENT` errors should only occur for missing files and "virtual files" (intermediate files created by plugins). It makes sense to ignore errors for virtual files, but the question is how to differentiate missing files from virtual ones.